037-kafka-canary-roadmap.md

```diff
@@ -0,0 +1,109 @@
+
+* Support for SASL `OAUTHBEARER` mechanism. Sarama supports this.
+* Support for forcing leader elections. Sarama doesn't support this.
+* Support for transactional produce. Sarama doesn't seem to support this, at least not in a usable way. You can send the transaction-related requests, but there is no client library support akin to the `TransactionManager`, for ensuring the RPCs are sent correctly. 
+
+### Differential observations
+
+By running multiple producers and/or consumers, a canary could more accurately indicate where problems lay. 
+
+For example, in a cluster with multiple listeners using different authentication mechanisms, if produce requests via a `SASL/OAUTHBEARER`-authenticated listener fail, but succeed via the `SASL/PLAIN`-authenticated listener you can infer that the problem is with the login mechanism. 
+
+There are many dimensions in which you can do this: 
+
+* DNS resolution
+* TLS (e.g. different versions or detecting imminent broker certificate expiry)
+* Producing to replicas on different volumes
+* With/without transactional producers, to flush out problems with transaction coordinators.
+* Similarly with/without using consumer groups. Using different group ids to detect problems with group coordinators.
+
+
+### Actionable documentation
+
+Assuming the differential observations feature proves to provide worthwhile information, we would proceed to also provide documented procedures recommending corrective actions based on the specific faults it reported.
+
+### Changes and removals
+
+* Logging mechanism will be changed to Log4j2, to have it consistent with other Strimzi components
+  * The log levels will have to be changed from `0, 1, 2` to `INFO, DEBUG, WARN, TRACE`.
+* `SARAMA_LOG_ENABLED` environment variable will be removed
+* `VERBOSITY_LOG_LEVEL` will be changed to `CANARY_LOG_LEVEL` and used inside the `log4j2.properties` file - default will be `INFO`
+* Implementation of dynamic watcher will be removed 
+  * For logging changes the dynamic reconfiguration will be implemented (similarly to operators repository)
+
+## Affected/not affected projects
+
+This affects the Canary project only.
+
+## Compatibility
+
+The API of the existing canary comprises:
+
+* The configuration env vars
+* Metrics (via the Prometheus scrape endpoint)
+* The status and health endpoints
+
+### Configuration
+While we can strive to minimize incompatibilities, it seems of questionable value.
+The Java clients are normally configured via `.properties` files.
+Providing a compatibility mapping from env var to properties would be a lot of work, given that the canary hasn't seem very wide usage.
+Some options would become obselete (e.g. `KAFKA_VERSION`), others are incompatible with how the Java clients work (e.g. `KAFKA_BOOTSTRAP_BACKOFF_MAX_ATTEMPTS`).
+So it seems better to provide a clean break.
+
+### Metrics
+
+It seems likely that metric names and labels can be maintained, so in that respect we could be compatible. 
```

Comment:
Link to the canary Golang metrics list would help https://github.com/strimzi/strimzi-canary#metrics-1
Are we taking all the metrics?